src/backend/optimizer/plan/setrefs.c

```diff
@@ -1079,16 +1078,6 @@ set_plan_refs(PlannerInfo *root, Plan *plan, int rtoffset)
 						newRL = lappend(newRL, rlist);
 					}
 					splan->returningLists = newRL;
-
-					/*
-					 * Set up the visible plan targetlist as being the same as
-					 * the first RETURNING list. This is for the use of
-					 * EXPLAIN; the executor won't pay any attention to the
-					 * targetlist.  We postpone this step until here so that
-					 * we don't have to do set_returning_clause_references()
-					 * twice on identical targetlists.
-					 */
-					splan->plan.targetlist = copyObject(linitial(newRL));
```

Comment:
Why don't you use the `set_returning_clause_references` function for `plan.targetlist `anymore?

Reply:
I think it probably makes sense to copy the `returningList` after `set_returning_clause_references` since it is used later.